sql/catalyst/src/main/scala/org/apache/spark/sql/util/SchemaUtils.scala

```diff
@@ -317,6 +317,8 @@ private[spark] object SchemaUtils {
       StructType(fields.map { field =>
         field.copy(dataType = replaceCollatedStringWithString(field.dataType))
       })
+    case CharType(l) => CharType(l)
+    case VarcharType(l) => VarcharType(l)
     case _: StringType => StringType
```

Comment:
can't we make this only the change in this line where we would say if it has no constraint than exchange